zero_bin/common/src/prover_state/persistence.rs

```diff
@@ -23,6 +25,19 @@ const VERIFIER_STATE_FILE_PREFIX: &str = "verifier_state";
 const ZK_EVM_CACHE_DIR_NAME: &str = "zk_evm_circuit_cache";
 const ZK_EVM_CACHE_DIR_ENV: &str = "ZK_EVM_CACHE_DIR";
 
+/// We version serialized circuits by the kernel hash they were serialized with,
+/// but we really only need a few of the starting hex nibbles to reliably
+/// differentiate.
+const KERNEL_HASH_PREFIX: usize = 8;
+
+/// When we serialize/deserialize circuits, we rely on the hash of the plonky
+/// kernel to determine if the circuit is compatible with our current binary. If
+/// the kernel hash of the circuit that we are loading in from disk differs,
+/// then using these circuits could potentially lead to incorrect results (but
+/// most likely just a crash).
```

Comment:
```suggestion
/// then using these circuits would cause failures during proof generation
```

Reply:
(i.e. this isn't potentially causing issues, this is _guaranteed_ to break proving)

Reply:
Yeah, lets be more assertive. :upside_down_face: